nomad/structs/network.go

```diff
@@ -113,6 +113,15 @@ func (idx *NetworkIndex) AddAllocs(allocs []*Allocation) (collide bool) {
 		}
 
 		if alloc.AllocatedResources != nil {
+			// Add network resources that are at the task group level
+			if len(alloc.AllocatedResources.Shared.Networks) > 0 {
+				for _, network := range alloc.AllocatedResources.Shared.Networks {
+					if idx.AddReserved(network) {
+						collide = true
```

Comment:
Not your code, but I found it a bit confusing that we track collisions, but callers seem to ignore value.  Is the return value used anywhere?

Reply:
Its used in the plan applier which checks that allocations fit on the node - https://github.com/hashicorp/nomad/blob/ea5fd244f65c8b8610c5a663fe3cdd5c0c371be7/nomad/structs/funcs.go#L102

Used because there could be a race between two different workers allocating the same node's network such that it results in a collision.